### PR TITLE
Feature/add subdirectory for answersfile

### PR DESCRIPTION
**Pull request by nilsdebruin**
The following test has been added to be able to validate if a subdirectory for an answersfile is possible

### CODE REVIEW

File: tests/test_cli.py [hunk 1]
```diff
@@ -39,6 +61,39 @@ def test_good_cli_run(tmp_path, template_path):
     assert answers["_src_path"] == str(template_path)
 
 
+def test_good_cli_run_dot_config(tmp_path, template_path_with_dot_config):
+    with local.cwd(str(template_path_with_dot_config)):
+        git_commands()
+
+    run_result = CopierApp.run(
+        ["--quiet", str(template_path_with_dot_config), str(tmp_path)],
```
Comment (yajo) [inline tests/test_cli.py:69]: You need this for the test to be correct.

```suggestion
        ["--quiet", "-a", ".config/altered-answers.yml", str(template_path_with_dot_config), str(tmp_path)],
```

**Review by yajo — CHANGES_REQUESTED** [review]
There's a little bug in the test code. Other than that, this seems like a good starting point to develop the solution! 😉 

Could you please mark the PR as draft in the mean time?

**Comment by nilsdebruin** [comment]
@yajo, many thanks for your check and feedback!

**Comment by nilsdebruin** [comment]
I have added a new approach to solve the issue

File: copier/main.py [hunk 2]
```diff
@@ -558,6 +558,13 @@ def _render_path(self, relpath: Path) -> Optional[Path]:
             if not part:
                 return None
             rendered_parts.append(part)
+
+        # The following logic was added based on the discussion: https://github.com/copier-org/copier/discussions/859
```
Comment (yajo) [inline copier/main.py:562]: This is not really needed, `git blame` will give you that!

If you want, you can add that kind of comment in the commit message though.

File: copier/main.py [hunk 3]
```diff
@@ -558,6 +558,13 @@ def _render_path(self, relpath: Path) -> Optional[Path]:
...
+        if len(rendered_parts) > 1 and str(self.answers_relpath).startswith(
+            rendered_parts[0]
+        ):
+            rendered_parts.pop(0)
```
Comment (yajo) [inline copier/main.py:566]: This also doesn't seem like the fix you're looking for. If there were more files under `.config/`, they'd be affected too, right?

File: copier/main.py (hunk 2 above)
Comment (nilsdebruin) [reply copier/main.py:562]: Ok, I will remove this!

File: copier/main.py (hunk 3 above)
Comment (nilsdebruin) [reply copier/main.py:566]: Good point, I will see if I can make it more specific tailored to the answersfile!

File: copier/main.py (hunk 2 above)
Comment (nilsdebruin) [reply copier/main.py]: Done!

File: copier/main.py (hunk 3 above)
Comment (nilsdebruin) [reply copier/main.py]: I have changed the implementation and added more tests, so different config folder levels are tested as well

**Comment by yajo** [comment]
Can you fix the conflicts please? So I can trigger the ci

**Comment by nilsdebruin** [comment]
> Can you fix the conflicts please? So I can trigger the ci

Done!

**Comment by yajo** [comment]
It took a bit longer because I couldn't understand why the fix worked. I pushed another commit with some cleanup and a useful comment for my future me. Merging, thanks!

**Pull request merged**